addons/common/functions/fnc_runQTE.sqf

```diff
@@ -0,0 +1,103 @@
+#include "..\script_component.hpp"
+#include "\a3\ui_f\hpp\definedikcodes.inc"
+/*
+ * Author: john681611
+ * Runs a Quick time Event.
+ *
+ * Arguments:
+ * 0: Object QTE is attached to <OBJECT>
+ * 1: remoteExec command to execute <STRING>
+ * 2: remoteExec command parameters <ARRAY>
+ * 3: remoteExec command target <STRING>
+ * 4: QTE Code <ARRAY>
+ * 5: Max interaction distance from attached object <NUMBER> (default: 10) 
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [car, "hint", "Yay", 1, ["↑", "↓", "→", "←"], 15] call ace_common_fnc_runQTE
+ *
+ * Public: Yes
+ */
+
+params ["_veh", "_command", "_command_params", "_command_exec_code", "_keyCode", ["_max_distance", 10]];
+
+GVAR(QTEVeh) = _veh;
+GVAR(QTECommand) = _command;
+GVAR(QTECommandParams) = _command_params;
+GVAR(QTECommandExecCode) = _command_exec_code;
+GVAR(QTEMaxDistance) = _max_distance;
+GVAR(QTECode) = _keyCode;
+GVAR(QTEHistory) = [];
+
+hint format ["%1: \n %2", localize LSTRING(QTEInput), [GVAR(QTECode)] call FUNC(getFormattedQTECode)];
+
+findDisplay 46 displayAddEventHandler ["KeyUp", {
```

Comment:
Take a look at https://cbateam.github.io/CBA_A3/docs/files/events/fnc_addBISEventHandler-sqf.html. It will allow you to eliminate the global variables needed to pass arguments to the display EH.

Reply:
You also need to check if the display is valid.